Prevent /W3 from being added to MSVC builds

charles-lunarg opened the pull request:
CMake 3.15 added a new policy which prevents the /W3 flag from being added to MSVC builds. To enable it, we need to set the CMake policy CMP0092 to the 'new' behavior. Note: To see the effect, I had to delete my CMakeCache.

ci-tester-lunarg commented:
CI Vulkan-Loader build queued with queue ID 2756.

ci-tester-lunarg commented:
CI Vulkan-Loader build queued with queue ID 2756.

ci-tester-lunarg commented:
CI Vulkan-Loader build # 1914 running.

ci-tester-lunarg commented:
CI Vulkan-Loader build # 1914 passed.

juan-lunarg reviewed (changes requested):
`cmake_minimum_required(VERSION 3.17.2)` by setting the min to 3.17.2 you shouldn't need to enably the new policy.

charles-lunarg commented:
According to the docs, you do need to manually set it.

> Unlike many policies, CMake version 3.26.3 does not warn when this policy is not set and simply uses OLD behavior.

https://cmake.org/cmake/help/latest/policy/CMP0092.html

juan-lunarg commented:
> does not warn when this policy is not set and simply uses OLD behavior.

?

Yes it won't produce a warning. That doesn't mean you need to manually set it.

charles-lunarg commented:
What the heck! I was getting the /W3 being added until I added this to cmake. Now, when I remove the change and rebuild, its gone!

Had I never cleaned the CMake cache on this machine since we updated to CMake 3.17? Very possible. As such, this PR is completely unnecessary.
